src/protocol.rs

```diff
@@ -327,19 +202,17 @@ impl Participant {
         )?;
         Ok(keyshare_public.X)
     }
+    */
 
     /// If presign record is populated, then this participant is ready to issue
     /// a signature
     /// The `presign_identifier` globally and uniquely defines a session for
     /// the pre-signing.
     #[instrument(skip_all, err(Debug))]
-    pub fn sign(
-        &mut self,
-        presign_identifier: Identifier,
-        digest: sha2::Sha256,
-    ) -> Result<SignatureShare> {
+    pub fn sign(record: PresignRecord, digest: sha2::Sha256) -> Result<SignatureShare> {
```

Comment:
Update associated documentation. Also, should this live in `Participant` anymore, or be moved to `PresignRecord`?

Reply:
I made #251 to move this to `PresignRecord`, yeah. The API is really awkward right now :) Omar had proposed having `Sign` be its own ProtocolParticipant, but I think we might not want that for now because there aren't any `Message`s in signing, it's just the local operation -- the calling application has to figure out where to send the signature shares once they get them (e.g. since it's often not the computation parties who are supposed to end up with the sig).